### PR TITLE
Add collective team in the progress report

### PR DESCRIPTION
**Pull request by ibnesayeed**
* Adds `All` as a derived field in teams that reflects in the both the tables
* Changes `Authors/Reviewers/Analysts` format to `Authors+Reviewers+Analysts=All`
* Refactors stats calculation function without changing the logic

### CODE REVIEW

**Comment by ibnesayeed** [comment]
This PR does not make team members list per chapter unique, in case the same person is in more than one roles because that will mess the stats up and I am not sure which way is more desired.

File: .github/workflows/progress-tracker.yml [hunk 1]
```diff
@@ -129,6 +131,7 @@ jobs:
                   team.Authors = parseUserNames(authors)
                   team.Reviewers = parseUserNames(reviewers)
                   team.Analysts = parseUserNames(analysts)
+                  team.All = [...team.Authors, ...team.Reviewers, ...team.Analysts]
```
Comment (rviscomi) [inline .github/workflows/progress-tracker.yml:134]: I think unique count of total contributors would be more useful. Could do something like `Array.from(new Set([...]))`.

File: .github/workflows/progress-tracker.yml [hunk 2]
```diff
@@ -139,7 +142,7 @@ jobs:
             }
 
             const formatTeam = team => {
-              return Object.entries(team).map(([k, v]) => `<span title="${k}: ${v.length ? v.join(', ') : 'TBD'}">${v.length}</span>`).join('/')
+              return Object.entries(team).map(([k, v]) => `<span title="${k}: ${v.length ? v.join(', ') : 'TBD'}">${v.length}</span>`).join('+').replace(/\+([^+]*)$/, '=$1')
```
Comment (rviscomi) [inline .github/workflows/progress-tracker.yml:145]: How about keeping "A/B/C" format and dealing with the fact that they won't add up to the unique total by formatting it as "A/B/C (D total)"?

File: .github/workflows/progress-tracker.yml (hunk 1 above)
Comment (ibnesayeed) [reply .github/workflows/progress-tracker.yml:134]: I actually wrote the code for making it unique, but reverted it for a few reasons:

* I was not sure how often a single person can be in more than one roles in the same chapter
* The sum can look odd in situations like `1+2+2=3`, but this is a representational thing that can be done differently
* In the Contributor details section, numbers will represent only one entry per chapter

That said, if you think that would be a better representation then I can bring my change back that I put together to make it unique initially.

File: .github/workflows/progress-tracker.yml (hunk 2 above)
Comment (ibnesayeed) [reply .github/workflows/progress-tracker.yml]: I have changed it slightly differently to keep it logically correct while still being compact.

File: .github/workflows/progress-tracker.yml [hunk 3]
```diff
@@ -72,17 +73,18 @@ jobs:
...
               nums.sort((a, b) => a - b)
-              const half = Math.floor(nums.length / 2)
+              const sum = nums.reduce((a,b) => a + b, 0)
               return [
-                nums.reduce((a,b) => a + b, 0),
-                Math.min(...nums),
-                Math.max(...nums),
-                (nums.reduce((a,b) => a + b, 0) / nums.length).toFixed(2),
-                nums[half]
+                sum,
```
Comment (rviscomi) [inline .github/workflows/progress-tracker.yml:83]: Just realized this doesn't work for unique contributors.

For example, in https://github.com/bazzadp/almanac.httparchive.org/issues/73 there are only 2 people (you and Barry) but the All/Total cell says 4. I'd expect this value to also deduplicate repeated contributors.

Comment (ibnesayeed) [reply .github/workflows/progress-tracker.yml:83]: You don't want to deduplicate there as that is a chapter-wise stats and it represents a different quantity. The summary table is not showing in the Total columns what you feel it should. It has always been different. For example, even before we introduced `All`, `Reviewers` were adding up to 3, though there were only two unique people involved.

Comment (ibnesayeed) [reply .github/workflows/progress-tracker.yml:83]: In short, we are deduplication multiple roles of a person in a single chapter, but we are not deduplicating roles of individuals across chapters, and we should not. If you are looking for an absolute number of unique contributors and in how many chapters they each contributed in one or more roles, the place you should be looking at is under the Contributor Details under the the All section.

**Pull request merged**